sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1203,14 +1203,24 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
   }
 
   test("alter table: recover partitions (sequential)") {
-    withSQLConf(RDD_PARALLEL_LISTING_THRESHOLD.key -> "10") {
+    val oldRddParallelListingThreshold = spark.sparkContext.conf.get(
+      RDD_PARALLEL_LISTING_THRESHOLD)
+    try {
+      spark.sparkContext.conf.set(RDD_PARALLEL_LISTING_THRESHOLD.key, "10")
```

Comment:
I have the impression that Spark conf is immutable. Are you sure this works?

Reply:
Yes this works. I have verified this by printing the value of this conf inside the main code in AlterTableRecoverPartitions class. Both of the tests prints different values - 0 and 10.

Also there are many tests which modify conf using "sc.conf.set". Ex - TaskSetManagerSuite, RDDSuite etc.